crates/orchestrator/src/jobs/state_update_job/mod.rs

```diff
@@ -226,10 +229,14 @@ impl StateUpdateJob {
             let onchain_data_size = 0;
             settlement_client.update_state_calldata(program_output, onchain_data_hash, onchain_data_size).await?
         } else if snos.use_kzg_da == Felt252::ONE {
-            // TODO: Build the blob & the KZG proof & send them to update_state_blobs
-            let kzg_proof = self.fetch_kzg_proof_for_block(block_no, fetch_from_tests).await;
-            let kzg_proof: [u8; 48] = kzg_proof.try_into().expect("kzg proof size must be 48 bytes");
-            settlement_client.update_state_blobs(vec![], kzg_proof).await?
+            let (blob_data, kzg_proof) = build_kzg_proof(block_no, fetch_from_tests).await?;
```

Comment:
fetching from s3 can be the responsibility of this function. build_kzg_proof will just calculate the proof. it will also help in moving common code outside when we move build_proof inside the DA client